src/langchain_google_alloydb_pg/alloydb_chat_message_history.py

```diff
@@ -15,35 +15,86 @@
 from __future__ import annotations
 
 import json
-from typing import List
+from typing import List, Sequence
 
 from langchain_core.chat_history import BaseChatMessageHistory
-from langchain_core.messages import BaseMessage, message_to_dict, messages_from_dict
+from langchain_core.messages import BaseMessage, messages_from_dict
 
 from .alloydb_engine import AlloyDBEngine
 
 
+async def _aget_messages(
+    engine: AlloyDBEngine, session_id: str, table_name: str
+) -> List[BaseMessage]:
+    """Retrieve the messages from AlloyDB"""
+    query = f"""SELECT data, type FROM "{table_name}" WHERE session_id = :session_id ORDER BY id;"""
+    results = await engine._afetch(query, {"session_id": session_id})
+    if not results:
+        return []
+
+    items = [{"data": result["data"], "type": result["type"]} for result in results]
+    messages = messages_from_dict(items)
+    return messages
+
+
 class AlloyDBChatMessageHistory(BaseChatMessageHistory):
     """Chat message history stored in an AlloyDB."""
 
-    def __init__(self, engine: AlloyDBEngine, session_id: str, table_name: str):
+    __create_key = object()
+
+    def __init__(
+        self,
+        key,
+        engine: AlloyDBEngine,
+        session_id: str,
+        table_name: str,
+        messages: List[BaseMessage],
+    ):
+        if key != AlloyDBChatMessageHistory.__create_key:
+            raise Exception(
+                "Only create class through 'create' or 'create_sync' methods!"
+            )
         self.engine = engine
         self.session_id = session_id
         self.table_name = table_name
+        self.messages = messages
 
-    @property
-    def messages(self) -> List[BaseMessage]:  # type: ignore
-        """Retrieve the messages from AlloyDB"""
-        query = f"""SELECT data, type FROM "{self.table_name}" WHERE session_id = :session_id ORDER BY id;"""
-        results = self.engine._run_as_sync(
-            self.engine._afetch(query, {"session_id": self.session_id})
-        )
-        if not results:
-            return []
+    @classmethod
+    async def create(
+        cls,
+        engine: AlloyDBEngine,
+        session_id: str,
+        table_name: str,
+    ):
+        table_schema = await engine._aload_document_table(table_name)
```

Comment:
should this be `engine._aload_chat_message_history_table(table_name)` instead of document table?

Reply:
Updated to be a generic load table function name